src/main/java/com/mineboundteam/minebound/magic/UtilitySpells/FireUtilitySpell.java

```diff
@@ -0,0 +1,186 @@
+            }
+        }
+    }
+
+    @Override
+    public void appendHoverText(ItemStack pStack, @Nullable Level pLevel, List<Component> pTooltipComponents, TooltipFlag pIsAdvanced) {
+        super.appendHoverText(pStack, pLevel, pTooltipComponents, pIsAdvanced);
+        pTooltipComponents.add(new TextComponent("While equipped in a ").withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent("utility slot").withStyle(ChatFormatting.DARK_PURPLE))
+                .append(":"));
+        double rateInSeconds = damageRate / 20d;
+        pTooltipComponents.add(new TextComponent("  - To all entities in an ").withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent("AOE of " + aoeRange + " blocks").withStyle(ChatFormatting.DARK_GREEN))
+                .append(", deals ")
+                .append(new TextComponent(new DecimalFormat("0.#").format(damage) + " heart" + (damage != 1 ? "s" : "") + " of fire damage").withStyle(ChatFormatting.RED))
+                .append(" every ")
+                .append(new TextComponent(new DecimalFormat("0.#").format(rateInSeconds) + " second" + (rateInSeconds > 1 ? "s" : "")).withStyle(ChatFormatting.DARK_GREEN)));
+        pTooltipComponents.add(new TextComponent("  - Gives fire immunity").withStyle(ChatFormatting.GRAY));
+        pTooltipComponents.add(new TextComponent("Multiple copies stack to increase the ").withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent("range").withStyle(ChatFormatting.DARK_GREEN))
+                .append(", ")
+                .append(new TextComponent("damage").withStyle(ChatFormatting.RED))
+                .append(", and ")
+                .append(new TextComponent("Mana cost").withStyle(manaColorStyle)));
+        pTooltipComponents.add(new TextComponent("Costs ").withStyle(ChatFormatting.GRAY)
+                .append(new TextComponent(manaCost + " Mana").withStyle(manaColorStyle)).append(" per entity damaged"));
+
+        LocalPlayer player = Minecraft.getInstance().player;
+        if (player != null) {
+            MutableComponent active = new TextComponent("Fire AOE is currently ").withStyle(ChatFormatting.UNDERLINE).withStyle(ChatFormatting.BOLD);
+            Optional<PlayerUtilityToggleProvider.UtilityToggle> toggle = player.getCapability(PlayerUtilityToggleProvider.UTILITY_TOGGLE).resolve();
+            if (toggle.isPresent() && toggle.get().fire)
+                pTooltipComponents.add(active.append(new TextComponent("ON").withStyle(ChatFormatting.GREEN)));
+            else
+                pTooltipComponents.add(active.append(new TextComponent("OFF").withStyle(ChatFormatting.RED)));
+        }
+    }
+
+    public static class FireUtilitySpellConfig implements IConfig {
+        public ForgeConfigSpec.IntValue MANA_COST;
+        public ForgeConfigSpec.IntValue AOE_RANGE;
+        public ForgeConfigSpec.DoubleValue DAMAGE;
+        public ForgeConfigSpec.IntValue DAMAGE_RATE;
```

Comment:
not sure if we need to give players _that_ much control

Reply:
I think if it's specified in the tooltip, it should probably be configurable.

Reply:
fair enough